app/assets/javascripts/discourse/app/components/footer-nav.gjs

```diff
@@ -0,0 +1,203 @@
+import Component from "@glimmer/component";
+import { tracked } from "@glimmer/tracking";
+import { action } from "@ember/object";
+import didInsert from "@ember/render-modifiers/modifiers/did-insert";
+import willDestroy from "@ember/render-modifiers/modifiers/will-destroy";
+import { service } from "@ember/service";
+import { modifier as modifierFn } from "ember-modifier";
+import DButton from "discourse/components/d-button";
+import concatClass from "discourse/helpers/concat-class";
+import htmlClass from "discourse/helpers/html-class";
+import { postRNWebviewMessage } from "discourse/lib/utilities";
+import { SCROLLED_UP, UNSCROLLED } from "discourse/services/scroll-direction";
+import { bind } from "discourse-common/utils/decorators";
+import not from "truth-helpers/helpers/not";
+
+class FooterNav extends Component {
+  @service appEvents;
+  @service capabilities;
+  @service scrollDirection;
+  @service composer;
+  @service modal;
+
+  @tracked shouldToggleMobileFooter = false;
+  @tracked canGoBack = false;
+  @tracked canGoForward = false;
+
+  currentRouteIndex = 0;
+  routeHistory = [];
+  backForwardClicked = false;
+
+  registerAppEvents = modifierFn(() => {
+    this.appEvents.on("page:changed", this, "_routeChanged");
+
+    return () => {
+      this.appEvents.off("page:changed", this, "_routeChanged");
+    };
+  });
+
+  @bind
+  registerDirectionObserver() {
+    this.scrollDirection.addObserver(
+      "lastScrollDirection",
+      this.toggleMobileFooter
+    );
+  }
+
+  @bind
+  unregisterDirectionObserver() {
+    this.scrollDirection.removeObserver(
+      "lastScrollDirection",
+      this.toggleMobileFooter
+    );
+  }
```

Comment:
This is using Ember's classic observer system. I think we should refactor to avoid it, and use native getters instead.

Like


```
get isVisible(){
  return [UNSCROLLED, SCROLLED_UP].includes(
    this.scrollDirection.lastScrollDirection
  ) && !this.composer.isOpen;
}
```

Reply:
Ah yes sure, I thought this is how you intended us to use the service